### ISSUE TITLE
Report inferred triggers

### ISSUE THREAD

**Issue by fpoli** [viperproject/silicon#586]
Is there a way to report which triggers has been inferred by Silicon? Dafny shows them with a tooltip in the IDE, or in the standard output if a special flag is enabled.

**Comment by mschwerhoff** [comment]
Currently not. Silicon merely calls Viper's trigger inference, though, which Prusti could as well. Not convenient, but might suffice for the time being.

@ArquintL Hovering over Viper quantifiers to reveal the triggers Viper will infer sounds like a nice feature.

**Comment by fpoli** [comment]
Thanks! Do you remember why Silicon implements a `TriggerGenerator` when Silver also provides a `TriggerGeneration`? I'm trying to understand if there is already a viper-to-viper transformation in Silver that can be called to add the inferred triggers to the AST of a program.

**Comment by mschwerhoff** [comment]
Good point, I had not thought about Silicon's own trigger generator: it shares the general approach with Viper, but is for Silicon's internal representation ("terms", basically SMT expressions). Some quantifiers on that level are constructed from Viper expressions, but need their own triggers (often only as a fall-back option).
